New build warnings with gcc 11.2.0

### What happened?

Getting some new warnings when compiling the latest master (after the merge of https://github.com/gnuradio/gnuradio/pull/5028)

### System Information

Ubuntu 21.10 with gcc 11.2.0

Build from source, tip of master branch.

### GNU Radio Version

3.10-git (master)

### Specific Version

git commit eab2ea6102fd7c84c9fd507b983698aa187df83b

### Steps to Reproduce the Problem

Build from source with gcc 11.2.0

### Relevant log output

```shell
In file included from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/buffer.h:15,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/buffer_double_mapped.h:15,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/io_signature.h:17,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/basic_block.h:15,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/block.h:17,
                 from /home/ubuntu/xfer/gnuradio/gr-soapy/python/soapy/bindings/../../../include/gnuradio/soapy/block.h:12,
                 from /home/ubuntu/xfer/gnuradio/gr-soapy/python/soapy/bindings/sink_python.cc:27:
/home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/custom_lock.h:26:7: warning: ‘class gr::custom_lock_if’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
   26 | class custom_lock_if
      |       ^~~~~~~~~~~~~~
In file included from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/buffer_double_mapped.h:15,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/io_signature.h:17,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/basic_block.h:15,
                 from /home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/block.h:17,
                 from /home/ubuntu/xfer/gnuradio/gr-soapy/python/soapy/bindings/../../../include/gnuradio/soapy/block.h:12,
                 from /home/ubuntu/xfer/gnuradio/gr-soapy/python/soapy/bindings/sink_python.cc:27:
/home/ubuntu/xfer/gnuradio/gnuradio-runtime/lib/../include/gnuradio/buffer.h:67:22: warning: base class ‘class gr::custom_lock_if’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
   67 | class GR_RUNTIME_API buffer : public custom_lock_if
      |                      ^~~~~~
```
```

--- a/gnuradio-runtime/include/gnuradio/custom_lock.h
+++ b/gnuradio-runtime/include/gnuradio/custom_lock.h
@@ -26,6 +26,8 @@ namespace gr {
 class custom_lock_if
 {
 public:
+    virtual ~custom_lock_if(){};
+
     /*!
      * This function will be executed on construction of the custom lock.
      */

